invenio_rdm_records/services/pids/manager.py

```diff
@@ -47,6 +47,36 @@ def _validate_pids_schemes(self, pids):
         if unknown_schemes:
             raise PIDSchemeNotSupportedError(unknown_schemes)
 
+    def _validate_identifiers(self, pids, errors):
+        """Validate and normalize identifiers."""
+        # TODO: Refactor to get it injected instead.
+        configured_identifiers = current_app.config["RDM_PERSISTENT_IDENTIFIERS"]
+
+        scheme_ids = []
+        for scheme, pids_attrs in pids.items():
+            id_ = pids_attrs.get("identifier")
+            if not id_:
+                continue
+
+            configured_identifier = configured_identifiers.get(scheme, {})
+            validator = configured_identifier.get("validator", lambda x: True)
+            normalizer = configured_identifier.get("normalizer", lambda x: x)
+            label = configured_identifier.get("label", scheme)
+
+            if not validator(id_):
+                errors.append(
+                    {
+                        "field": f"pids.{scheme}",
+                        "messages": [_("Invalid {scheme}").format(scheme=label)],
+                    }
+                )
+            else:
+                scheme_ids.append((scheme, normalizer(id_)))
+
+        # Modify outside of iteration - updates the pids globally
+        for scheme, id_ in scheme_ids:
+            pids[scheme]["identifier"] = id_
+
```

Comment:
I was sitting on this change for a little while, so might as well pass it in here when looking at how DOI minting was affected by missing publisher.

I think it makes the code more readable. 

_Aside_
Also, it reveals how we could go one-step further in my ideal world: 

- make the `"RDM_PERSISTENT_IDENTIFIERS"`config take a factory object instead of a dict 
- that returns `ConfiguredIdentifier` objects that have a `label`, `scheme`, `validate`, `normalize` interface.

Basically an extract class to solve the primitive obsession on `dict` and reduce the config interface size. Because more changes involved, I'd be happy to submit a separate PR, it that sounds interesting.